trivup/apps/OauthbearerOIDCApp.py

```diff
@@ -0,0 +1,231 @@
+        --url localhost:8000/retrieve \
+        -H "Accept: application/json" \
+        -H "Content-type: application/x-www-form-urlencoded" \
+        -H "Authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK \
+            (base64 string generated from CLIENT_ID:CLIENT_SECRET)"
+        -d "grant_type=client_credentials,scope=test-scope"
+        """
+        if self.headers.get('Content-Length', None) is None:
+            self.send_error(400, 'Content-Length field is required')
+            return
+
+        if self.headers.get('Content-type', None) is None:
+            self.send_error(400, 'Content-type field is required')
+            return
+
+        content_length = int(self.headers['Content-Length'])
+        post_data = self.rfile.read(content_length)
+
+        if self.headers.get('Authorization', None) is None:
+            self.send_error(400, 'Authorization field is required')
+            return
+
+        if self.headers.get('Accept', None) != "application/json":
+            self.send_error(400, 'Accept field should be "application/json"')
+            return
+
+        if post_data is None:
+            self.send_error(400,
+                            'grant_type=client_credentials and scope \
+                             fields are required in data')
+            return
+
+        self.send_response(200)
+        self.send_header('Content-type', 'application/json')
+        self.end_headers()
+        token = generate_token(4)
+        self.wfile.write(json.dumps(token, indent=4).encode())
+
+    def generate_badformat_token_for_client(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'application/json')
+        self.end_headers()
+
+        token = generate_token(30, False)
+        self.wfile.write(json.dumps(token, indent=4).encode())
+
+    def generate_expired_token_for_client(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'application/json')
+        self.end_headers()
+
+        token = generate_token(-1)
+        self.wfile.write(json.dumps(token, indent=4).encode())
+
+    def do_POST(self):
+        if self.path.endswith("/retrieve"):
+            self.generate_valid_token_for_client()
+        elif self.path.endswith("/retrieve/badformat"):
+            self.generate_badformat_token_for_client()
+        elif self.path.endswith("/retrieve/expire"):
+            self.generate_expired_token_for_client()
+        else:
+            self.send_error(404, 'URL is not valid: %s' % self.path)
+
+
+class OauthbearerOIDCHttpServer():
+    def run_http_server(self, port):
+        server = HTTPServer(('localhost', port), WebServerHandler)
+        server.serve_forever()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Trivup Oauthbearer OIDC \
+                                                  HTTP server')
+    parser.add_argument('--port', type=int, dest='port',
+                        default=False, required=True,
```

Comment:
default should hold the default value (and this is an int, not a bool), but since the argument is required there should be no default, just remove the param.